### ISSUE TITLE
Null pointer exception thrown when opening a job config and useSpecs==true

### ISSUE THREAD

**Issue by rompic** [jfrog/jenkins-artifactory-plugin#407]
**Describe the bug**
Null pointer exception thrown when opening a freestyle job config and useSpecs==true

**To Reproduce**
Open a freestyle Job which uses Specs without configuring Legacy patterns

**Expected behavior**
No exception in the log

**Screenshots**
See exception below

**Versions**
- Jenkins Artifactory plugin version: 3.10.3
- Jenkins operating system: Windows 10
- Artifactory Version: 7.10.6
- Jenkins 2.263.1 (LTS)

**Additional context**
[exception.txt](https://github.com/jfrog/jenkins-artifactory-plugin/files/5762098/exception.txt)

**Comment by yahavi** [comment]
@rompic 
Thanks for reporting this issue. 
To improve compatibility with the Jenkins core for the current and future versions, we decided to start removing Javascript code and use the Jenkins Jelly whenever it is possible. 

To resolve this issue, in #409 we removed the support for legacy patterns from the UI. 
Here is a snapshot that should resolve this issue: 
https://oss.jfrog.org/artifactory/oss-snapshot-local/org/jenkins-ci/plugins/artifactory/3.10.x-SNAPSHOT/artifactory-3.10.x-20210105.082253-46.hpi

We'd appreciate your feedback on that.
